pdf_generator/tex_generation/template_engine.py

```diff
@@ -119,42 +81,146 @@ def create_jinja_environment(templates_to_use='default'):
     return environment
 
 
-def plugin_name(name):
-    return ' '.join((part.title() for part in name.split('_')))
+def get_five_longest_entries(summary, top=5):
+    sorted_summary = dict()
+    if len(summary) < 6:
+        return summary
+    for key in sorted(summary, key=lambda key: len(summary[key]), reverse=True):
+        sorted_summary.update({key: summary[key]})
+        if len(sorted_summary) == top:
+            return sorted_summary
+    return sorted_summary
+
+
+def exploit_mitigation(summary):
+    summary = summary['exploit_mitigations']['summary']
+    max_count = count_mitigations(summary)  # bar is maxed at 6
+    pie_num, canary_num, relro_num, nx_num, fortify_num = 0, 0, 0, 0, 0
+    for selected_summary in summary:
+        if 'PIE' in selected_summary and 'present' in selected_summary:
+            pie_num += len(summary[selected_summary])
+        if 'RELRO' in selected_summary and 'enabled' in selected_summary:
+            relro_num += len(summary[selected_summary])
+        if 'Canary' in selected_summary and 'enabled' in selected_summary:
+            canary_num += len(summary[selected_summary])
+        if 'NX' in selected_summary and 'enabled' in selected_summary:
+            nx_num += len(summary[selected_summary])
+        if 'FORTIFY' in selected_summary and 'enabled' in selected_summary:
+            fortify_num += len(summary[selected_summary])
+    return '{0}{2}/{3}{1},{0}{4}/{5}{1},' \
+           '{0}{6}/{7}{1},{0}{8}/{9}{1},' \
+           '{0}{10}/{11}{1}'.format('{', '}', 'CANARY', canary_num * 6 / max_count, 'PIE', pie_num * 6 / max_count,
+                                    'RELRO', relro_num * 6 / max_count, 'NX', nx_num * 6 / max_count,
+                                    'FORTIFY\_SOURCE', fortify_num * 6 / max_count)
+
+
+def count_mitigations(summary):
+    for mitigation in ['Canary', 'NX', 'RELRO', 'PIE', 'FORTIFY']:
+        count = count_this_mitigation(summary, mitigation)
+        if count != 0:
+            return count
+    return count
+
+
+def count_this_mitigation(summary, mitigation):
+    count = 0
+    for selected_summary in summary:
+        if mitigation in selected_summary:
+            count += len(summary[selected_summary])
+    return count
+
+
+def software_components(software_string):
+    software = software_string
+    ver_number = ''
+    if ' ' in software_string:
+        if len(software_string.split(' ')) > 2:
+            software = ''.join(software_string.split(' ')[:-1])
```

Comment:
`software_string.split(' ')` is called in six places -> should be stored in a variable